### ISSUE TITLE
mirtrace_species variable not defined

### ISSUE THREAD

**Issue by kopichris** [nf-core/smrnaseq#488]
### Description of the bug

According to the [docs](https://nf-co.re/smrnaseq/2.4.0/parameters#genome), the `--mirtrace_species` variable is set when the `--genome` variable is defined; however, setting the `--genome` parameter results in the following error:
```
Reference species for miRTrace is not defined via the `--mirtrace_species` parameter.
```
The error occurs because the `--mirtrace_species` variable is not globally defined, and is accessed by a function that does not have access to it in the [pipeline_initialisation](https://github.com/nf-core/smrnaseq/blob/master/subworkflows/local/utils_nfcore_smrnaseq_pipeline/main.nf) workflow:
```
def validateInputParameters() {
    genomeExistsError()

    if (!params.mirgenedb && !params.mirtrace_species) {
            error("Reference species for miRTrace is not defined via the `--mirtrace_species` parameter.")
    }
...
```
I am working on making a pull request to the `dev` branch if there are no issues!

### Command used and terminal output

```console
$ nextflow run main.nf -profile docker --input samplesheet.csv --outdir debug_test --genome "GRCh38"

 N E X T F L O W   ~  version 24.04.2

Launching `main.nf` [adoring_bell] DSL2 - revision: 96a86dc2ea

[-        ] process > PREPARE_GENOME:CLEAN_FASTA  -
[-        ] process > PREPARE_GENOME:INDEX_GENOME -
Reference species for miRTrace is not defined via the `--mirtrace_species` parameter.
```

### System information

Nextflow version: 24.04.2
Hardware: HPC
Executor: local
Container engine: Docker
OS: Linux
Version of nf-core/smrnaseq: 2.4.0

**Comment by nschcolnicov** [comment]
Hi @kopichris Thanks for opening this issue, I see that you are right. The `validateInput()` function is running before these parameters have a chance to be set with `params.mirtrace_species    = getGenomeAttribute('mirtrace_species')`. Please tag me on the PR once its ready

**Comment by kopichris** [comment]
Hi @nschcolnicov, that's what I suspected at first too, but I think it's just a parameter visibility issue.

It turns out that the `--mitrace_species` parameter is correctly defined and set prior to `validateInputParameters()` being called - this can be seen on line 31 in main.nf.

The parameter is then passed as input to the [prepare_genome](https://github.com/kopichris/smrnaseq/blob/master/subworkflows/local/prepare_genome/main.nf) workflow, where the following check happens:
```
if (!params.mirgenedb && !val_mirtrace_species) {
    exit 1, "Reference species for miRTrace is not defined via the --mirtrace_species parameter."
}
```
A similar check is subsequently performed within the [pipeline_initialisation](https://github.com/kopichris/smrnaseq/blob/master/subworkflows/local/utils_nfcore_smrnaseq_pipeline/main.nf) workflow, where the `validateInputParameters()` function is implemented:
```
if (!params.mirgenedb && !params.mirtrace_species) {
    error("Reference species for miRTrace is not defined via the `--mirtrace_species` parameter.")
}
```
The error occurs because the `params.mirtrace_species` parameter was not passed as input to the pipeline_initialization workflow, so `params.mirtrace_species` is `null` instead of `hsa`, for example.

I cannot see a reason why this logic should be implemented in two places: once in prepare_genome and once in pipeline_initialization.  I leaned towards removing it from the latter workflow because it was already being correctly handled in prepare_genome.  Thoughts @nschcolnicov @atrigila?

**Comment by nschcolnicov** [comment]
@kopichris I think that change makes sense, not sure why the same check is run twice, probably some leftover code that we missed when refactoring the pipeline. Please go ahead and create the PR. Thanks for looking into this! 😄

**Comment by atrigila** [comment]
Closed via #493 . Thank you for reporting this issue and the proposed solution :)
